gomock/call.go

```diff
@@ -70,11 +70,56 @@ func (c *Call) MaxTimes(n int) *Call {
 	return c
 }
 
-// Do declares the action to run when the call is matched.
+// DoAndReturn declares the action to run when the call is matched.
+// The return values from this function are returned by the mocked function.
+// It takes an interface{} argument to support n-arity functions.
+func (c *Call) DoAndReturn(f interface{}) *Call {
+	// TODO: Check arity and types here, rather than dying badly elsewhere.
+	v := reflect.ValueOf(f)
```

Comment:
Since the code is structured this way now we can verify that v.Kind() != reflect.Func and c.t.Fatalf if not; we can also explicitly enforce that len(args) == v.NumIn() below instead of panicking in reflect. Feel free to keep the latter as a todo/do it in a later commit.